File: src/main/java/sk/hudak/prco/dao/db/NewProductEntityDbDao.java
package sk.hudak.prco.dao.db;

import sk.hudak.prco.dao.BaseDao;
import sk.hudak.prco.dto.NewProductFilterUIDto;
import sk.hudak.prco.model.NewProductEntity;

import java.util.List;
import java.util.Optional;

public interface NewProductEntityDbDao extends BaseDao<NewProductEntity> {

    boolean existWithUrl(String url);

    List<NewProductEntity> findAll();

    List<NewProductEntity> findInvalid(int maxCountOfInvalid);

    long countOfAllInvalidNewProduct();

    Optional<NewProductEntity> findFirstInvalid();

    List<NewProductEntity> findByFilter(NewProductFilterUIDto filter);

    //TODO rename na countOfAllNewProducts
    long getCountOfAllNewProducts();
}

File: src/main/java/sk/hudak/prco/dao/db/NewProductEntityDbDao.kt
package sk.hudak.prco.dao.db

import sk.hudak.prco.api.EshopUuid
import sk.hudak.prco.dao.BaseDao
import sk.hudak.prco.dto.product.NewProductFilterUIDto
import sk.hudak.prco.model.NewProductEntity

interface NewProductEntityDbDao : BaseDao<NewProductEntity> {

    fun countOfAll(): Long

    fun existWithUrl(url: String): Boolean

    fun findAll(): List<NewProductEntity>

    fun findInvalid(maxCountOfInvalid: Int): List<NewProductEntity>

    fun countOfAllInvalidNewProduct(): Long

    fun findFirstInvalid(): NewProductEntity?

    fun findByFilter(filter: NewProductFilterUIDto): List<NewProductEntity>

    fun findByCount(eshopUuid: EshopUuid, maxCountToDelete: Long): List<NewProductEntity>

    fun countOfAllProductInEshop(eshopUuid: EshopUuid): Long

}
